neat_ml/bubblesam/bubblesam.py

```diff
@@ -0,0 +1,330 @@
+    and filtering out non-circular masks based on a circularity threshold.
+
+    Parameters
+    ----------
+    masks_summary_df : pd.DataFrame
+        DataFrame containing at least a column 'segmentation' with boolean masks.
+    area_threshold : float
+        Minimum area for the mask to be retained.
+    circularity_threshold : float
+        Circularity threshold to consider for the mask to be "circular."
+
+    Returns
+    -------
+    pd.DataFrame
+        A filtered DataFrame with new columns (contour, bbox, major_axis, minor_axis, 
+        area, radius, circ, euler_number) but excluding 'segmentation'.
+    """
+    filtered_rows = []
+
+    for idx, row in masks_summary_df.iterrows():
+        seg = row['segmentation']
+        labeled_seg = label(seg)
+        props_list = regionprops(labeled_seg)
+        if len(props_list) == 0:
+            continue
+
+        rp = props_list[0]
+        area = rp.area
+        perimeter = rp.perimeter
+        if perimeter == 0:
+            continue
+
+        circ = (4.0 * np.pi * area) / (perimeter ** 2)
+        major_axis = rp.major_axis_length
+        minor_axis = rp.minor_axis_length
+        if area >= area_threshold and circ >= circularity_threshold:
+            mask_info = {              
+                'bbox': rp.bbox,
+                'contour': find_contours(seg, level=0.5)[0],
+                'major_axis': major_axis,
+                'minor_axis': minor_axis,
+                'area': area,
+                'radius': np.sqrt(area / np.pi),
+                'circ': circ,
+                'euler_number': rp.euler_number
+            }
+
+            filtered_rows.append(mask_info)
+
+    df_filtered = pd.DataFrame(filtered_rows)
+    return df_filtered
+
+def plot_filtered_masks(
+    original_image: np.ndarray,
+    masks_summary_df: pd.DataFrame,
+    output_path: str
+):
+    """
+    Plots the filtered masks (contours + bounding boxes) on the original image 
+    and saves the resulting figure.
+
+    Parameters
+    ----------
+    original_image : np.ndarray
+        Original image array.
+    masks_summary_df : pd.DataFrame
+        DataFrame containing the columns 'contour' and 'bbox' for each mask.
+    output_path : str
+        File path to save the resulting figure.
+    """
+    fig, ax = plt.subplots(figsize=(10, 10))
+    ax.imshow(original_image, cmap="gray")
+
+    for idx, row in masks_summary_df.iterrows():
+        contour = row['contour']
+        bbox = row['bbox']
+        ax.plot(contour[:, 1], contour[:, 0], linewidth=1, color='blue')
+        min_row, min_col, max_row, max_col = bbox
+        rect = Rectangle(
+            (min_col, min_row),
+            max_col - min_col,
+            max_row - min_row,
+            linewidth=1,
+            edgecolor='green',
+            facecolor='none'
+        )
+        ax.add_patch(rect)
+    ax.axis("off")
+    plt.savefig(output_path, bbox_inches='tight')
```

Comment:
Here and elsewhere--let's try to use `fig`, `ax` objects instead of the "global" `plt` approach, for the usual reasons.